### PR TITLE
Implementation of the Transaction status API

### PR DESCRIPTION
**Pull request by itsyaasir**

### CODE REVIEW

**Review by collinsmuriuki — CHANGES_REQUESTED** [review]

**Comment by collinsmuriuki** [comment]
> I have just realized through out the services API , we have a typo in the Response struct and I just noticed it now. Take a look at the `OriginatorCoversationID`.
> 
> ```rust
>     #[serde(rename(deserialize = "OriginatorCoversationID"))]
>     pub originator_coversation_id: String,
> ```
> 
> Does safaricom API returns something like this ?

The API returns `OriginatorCoversationID`. If the value was incorrect, then deserialization would fail

**Comment by collinsmuriuki** [comment]
Good work 👍🏽

File: Cargo.toml [hunk 1]
```diff
@@ -32,12 +32,14 @@ default = [
...
+	"transaction_reversal",
+	"transaction_status"
 ]
 account_balance = ["dep:openssl", "dep:base64"]
 b2b = ["dep:openssl", "dep:base64"]
 b2c = ["dep:openssl", "dep:base64"]
 c2b_register = []
 c2b_simulate = []
 express_request = ["dep:chrono", "dep:base64"]
-transaction_reversal = ["dep:openssl", "dep:base64"]
```
[the remark this answers is not in the document]
Comment (itsyaasir) [reply Cargo.toml:43]: It was a mistake, I'll return them again.

[the remark this answers is not in the document]
Comment (itsyaasir) [reply Cargo.toml:43]: Done

**Comment by itsyaasir** [comment]
It's like this test behaves sporadically `stk_push_test::stk_push_test`

**Comment by collinsmuriuki** [comment]
Yes very annoying, the API on sandbox will randomly return an error response. Will address this soon though

**Comment by itsyaasir** [comment]
> Yes very annoying, the API on sandbox will randomly return an error response. Will address this soon though

You can approve this so we can merge it in and close the issue.

**Review by collinsmuriuki — APPROVED** [review]

**Pull request merged**